### DISCUSSION TITLE
Support VisionOS

### DISCUSSION THREAD

**Discussion by lucasrose** [expo/expo#26946] [Proposals and Ideas]
Hey team, I love your products and the work you've done to support the various platforms in the apple ecosystem. Any plans to support visionOS in the future (I saw callstack started a fork of react native https://github.com/callstack/react-native-visionos)?

**Comment by JulianKingman** [comment]
Cross-posting this video here to add to the discussion:
https://twitter.com/Baconbrix/status/1753927218697208102

I'd love to see visionOS incorporated into Expo, or addable as a native dependency

**Comment by tsapeta** [comment]
We've recently added support for writing Expo modules targeting macOS and tvOS. This was part of the effort to support visionOS eventually.

To support these two platforms we've made Expo Autolinking agnostic to Apple platforms, so now it's pretty straightforward to enable one more.
I believe that changes in the native code that were made for tvOS should be more or less compatible with visionOS as well, but we'll need to go through those sensitive parts that need platform-specific implementation again and test them. It's mostly just a matter of adding some ifs in the modules core and adding missing logic to Expo CLI so it works properly on other platforms.

That being said, yes, we're looking to support visionOS in our products. Stay tuned!
